### DISCUSSION TITLE
Improvements of Docker images (Helm home vars, non-root user, etc.)

### DISCUSSION THREAD

**Discussion by pathob** [helmfile/helmfile#296] [Ideas]
Hi @mumoshu @yxxhero 

currently, all Docker images will fail, when the `HOME` environment variable is set to a value other than `/root` as this is the location where the necessary Helm plugins (e.g. diff) are installed.

```
$ docker run -it -e HOME=/other ghcr.io/helmfile/helmfile:v0.145.3 helm diff
Error: unknown command "diff" for "helm"
Run 'helm --help' for usage.

$ docker run -it -e HOME=/other ghcr.io/helmfile/helmfile-debian-stable-slim:v0.145.3 helm diff
Error: unknown command "diff" for "helm"
Run 'helm --help' for usage.

$ docker run -it -e HOME=/other ghcr.io/helmfile/helmfile-ubuntu:v0.145.3 helm diff
Error: unknown command "diff" for "helm"
Run 'helm --help' for usage.
```

Having the `HOME` variable overridden is for example the case in all GitHub Actions (when using the container feature). In the Docker command, the variable is set like this: `-e "HOME=/github/home"`

```
name: Helmfile

on:
  push:
    branches: [ main ]

jobs:
  helmfile:
    runs-on: ubuntu-20.04

    container:
      image: ghcr.io/helmfile/helmfile:v0.145.3

    steps:
    - name: Test
      run: |
        helm env
```

Which results in:

```
HELM_BIN="helm"
HELM_CACHE_HOME="/github/home/.cache/helm"
HELM_CONFIG_HOME="/github/home/.config/helm"
HELM_DATA_HOME="/github/home/.local/share/helm"
HELM_DEBUG="false"
HELM_KUBEAPISERVER=""
HELM_KUBEASGROUPS=""
HELM_KUBEASUSER=""
HELM_KUBECAFILE=""
HELM_KUBECONTEXT=""
HELM_KUBETOKEN=""
HELM_MAX_HISTORY="10"
HELM_NAMESPACE="default"
HELM_PLUGINS="/github/home/.local/share/helm/plugins"
HELM_REGISTRY_CONFIG="/github/home/.config/helm/registry/config.json"
HELM_REPOSITORY_CACHE="/github/home/.cache/helm/repository"
HELM_REPOSITORY_CONFIG="/github/home/.config/helm/repositories.yaml"
```

To prevent this, it would be possible to set the Helm home variables in the Dockerfiles as follows:

```
ENV HELM_CACHE_HOME="/root/.cache/helm"
ENV HELM_CONFIG_HOME="/root/.config/helm"
ENV HELM_DATA_HOME="/root/.local/share/helm"
```

I can provide a PR and I would also like to discuss running the containers with a dedicated non-root user per default, but please give me some feedback about the Helm home variables first.

Thanks
Patrick

**Comment by pathob** [comment]
It might also be interesting to only do that for `HELM_DATA_HOME` as e.g. the `/github/home` directory in GitHub Actions provides some persistence during the whole workflow run, but I'm not familiar enough with Helm internals at this point.

**Comment by yxxhero** [comment]
@pathob just do it.
